src/ucp/core/ucp_ep.c

```diff
@@ -357,11 +362,14 @@ ucs_status_t ucp_worker_create_mem_type_endpoints(ucp_worker_h worker)
 void ucp_worker_destroy_mem_type_endpoints(ucp_worker_h worker)
 {
     ucs_memory_type_t mem_type;
+    ucp_ep_h ep;
 
     ucs_memory_type_for_each(mem_type) {
-        if (worker->mem_type_ep[mem_type] != NULL) {
-           ucp_ep_destroy_internal(worker->mem_type_ep[mem_type]);
-           worker->mem_type_ep[mem_type] = NULL;
+        ep = worker->mem_type_ep[mem_type];
+        if (ep != NULL) {
+            ucs_assert(ucp_ep_config(ep)->key.err_mode == UCP_ERR_HANDLING_MODE_NONE);
+            ucp_ep_destroy_internal(ep);
+            worker->mem_type_ep[mem_type] = NULL;
```

Comment:
can revert this

Reply:
hmm, assert could be useful...
reverted